FlowCrypt/Functionality/Services/Compose Message Service/ComposeMessageService.swift

```diff
@@ -83,27 +83,24 @@ final class ComposeMessageService {
             return .failure(.validationError(.missedPublicKey))
         }
 
-        switch getPubKeys(for: recipients) {
-        case .success(let allRecipientPubs):
-            let replyToMimeMsg = input.replyToMime
-                .flatMap { String(data: $0, encoding: .utf8) }
-
-            let msg = SendableMsg(
-                text: text,
-                to: recipients.map(\.email),
-                cc: [],
-                bcc: [],
-                from: email,
-                subject: subject,
-                replyToMimeMsg: replyToMimeMsg,
-                atts: atts,
-                pubKeys: allRecipientPubs + [myPubKey]
-            )
-
-            return .success(msg)
-        case .failure(let error):
-            return .failure(.validationError(error))
-        }
+        return getPubKeys(for: recipients)
+            .mapError { ComposeMessageError.validationError($0) }
+            .map { allRecipientPubs in
+                let replyToMimeMsg = input.replyToMime
+                    .flatMap { String(data: $0, encoding: .utf8) }
+
+                return SendableMsg(
+                    text: text,
+                    to: recipients.map(\.email),
+                    cc: [],
+                    bcc: [],
+                    from: email,
+                    subject: subject,
+                    replyToMimeMsg: replyToMimeMsg,
+                    atts: atts,
+                    pubKeys: allRecipientPubs + [myPubKey]
+                )
+            }
     }
```

Comment:
It's similarly weird to use `map` if we are talking about a single Future executing one time with a single failure or success. Map implies working with array of potentially many items, which seems a mismatch for what we're doing (even if this is otherwise common in iOS development).

I'll merge this but file a separate issue.